be/src/storage/lake/delta_writer.h

```diff
@@ -0,0 +1,57 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "common/statusor.h"
+#include "gutil/macros.h"
+
+namespace starrocks {
+class MemTracker;
+class SlotDescriptor;
+} // namespace starrocks
+
+namespace starrocks::vectorized {
+class Chunk;
+}
+
+namespace starrocks::lake {
+
+class DeltaWriterImpl;
```

Comment:
Why not implement the logic directly to `DeltaWriter`？

Reply:
Reduce c++ build times: https://www.lattix.com/reduce-c-build-times-part-2-with-the-pimpl-idiom/#:~:text=The%20big%20advantage%20is%20that,without%20recompiling%20the%20client%20code.